components/builder-depot/src/data_store.rs

```diff
@@ -133,20 +133,18 @@ impl PackagesIndex {
         Ok(val)
     }
 
-    pub fn list(&self,
-                id: &str,
-                offset: isize,
-                count: isize)
-                -> Result<Vec<depotsrv::PackageIdent>> {
+    pub fn list(&self, id: &str, start: isize, stop: isize) -> Result<Vec<depotsrv::PackageIdent>> {
         let conn = self.pool().get().unwrap();
-        match conn.zrange::<String, Vec<String>>(Self::key(&id.to_string()), offset, count) {
+
+        // Note: start and stop are INCLUSIVE ranges
+        match conn.zrange::<String, Vec<String>>(Self::key(&id.to_string()), start, stop) {
             Ok(ids) => {
                 // JW TODO: This in-memory sorting logic can be removed once the Redis sorted set
                 // is pre-sorted on write. For now, we'll do it on read each time.
                 let mut ids: Vec<package::PackageIdent> =
                     ids.iter().map(|id| package::PackageIdent::from_str(id).unwrap()).collect();
                 ids.sort();
-                ids.reverse();
+                // ids.reverse();
```

Comment:
@chefsalim the ids are reversed here because they don't come out in the order we expected. If you can verify the things are in the proper order with your changes here, then we're probably alright. Just make sure you test it with a large enough data set to verify that.

I'd remove the comment completely if we no longer need to call `reverse()`